contracts/src/everlong/EverlongERC4626.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+// import { IEverlongERC4626 } from "../interfaces/IEverlongERC4626.sol";
+import { ERC4626 } from "solady/tokens/ERC4626.sol";
+
+contract EverlongERC4626 is ERC4626 {
+    /// @notice Whether virtual shares will be used to mitigate the inflation attack.
+    bool public constant useVirtualShares = true;
+
+    /// @notice Used to reduce the feasibility of an inflation attack.
+    uint8 public constant decimalsOffset = 3;
+
+    /// @notice Address of the underlying Hyperdrive instance.
+    address internal immutable _underlying;
+
+    /// @notice Decimals used by the underlying Hyperdrive asset.
+    uint8 internal immutable _decimals;
+
+    /// @notice Name of the Everlong token.
+    string internal _name;
+
+    /// @notice Symbol of the Everlong token.
+    string internal _symbol;
+
+    constructor(
+        address underlying_,
+        string memory name_,
+        string memory symbol_
+    ) {
+        _underlying = underlying_;
+
+        (bool success, uint8 result) = _tryGetAssetDecimals(underlying_);
+        _decimals = success ? result : _DEFAULT_UNDERLYING_DECIMALS;
+
+        _name = name_;
+        _symbol = symbol_;
+    }
+
+    /// @dev Address of the underlying Hyperdrive instance.
+    ///
+    /// - MUST be an ERC20 token contract.
+    /// - MUST NOT revert.
+    function asset() public view virtual override returns (address) {
+        return _underlying;
+    }
```

Comment:
Hyperdrive isn't an ERC20 token contract. It may make more sense for this to be the idle liquidity. I was thinking of this recently, but Everlong may not fit that well into the ERC4626 standard because of withdrawal shares. Just keep that in mind.

Reply:
Gave this more thought, and totally agree. It seems like the `_underlying` variable should actually be the `baseToken` or `vaultSharesToken` for the underlying Hyperdrive instance since it will be used as the token for deposits/mints, idle liquidity, and bond purchases.

Reply:
That makes sense to me too. The big thing I'm unsure about is whether or not we can really support ERC4626 fully due to withdrawal shares, but we can circle back on that.